src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -46,6 +46,22 @@ public class ObservableCollection<T> : Collection<T>, INotifyCollectionChanged,
         /// <exception cref="ArgumentNullException"> collection is a null reference </exception>
         public ObservableCollection(IEnumerable<T> collection) : base(CreateCopy(collection)) { }
 
+        /// <summary>
+        /// Initializes a new instance of the ObservableCollection class
+        /// that contains elements copied from the specified list
+        /// </summary>
+        /// <param name="list">The list whose elements are copied to the new list.</param>
+        /// <remarks>
+        /// The elements are copied onto the ObservableCollection in the
+        /// same order they are read by the enumerator of the list.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException"> list is a null reference </exception>
+        public ObservableCollection(List<T> list)
+            : base((list != null) ? new List<T>(list.Count) : list)
+        {
+            CreateCopy(list);
+        }
```

Comment:
This constructor implementation looks incorrect. `CreateCopy` in corefx is a little different than the desktop implementation's `CopyFrom`...

I think it should be something like:

```c#
public ObservableCollection(List<T> list) : base(CreateCopy(list)) { }
```

Or, to use the right paramName depending on which constructor is called, it may be better to do something like:

```c#
public ObservableCollection(IEnumerable<T> collection) : base(CreateCopy(collection, nameof(collection))) { }

public ObservableCollection(List<T> list) : base(CreateCopy(list, nameof(list))) { }

private static List<T> CreateCopy(IEnumerable<T> collection, string paramName)
{
    if (collection == null)
        throw new ArgumentNullException(paramName);
 
    return new List<T>(collection);
}
```